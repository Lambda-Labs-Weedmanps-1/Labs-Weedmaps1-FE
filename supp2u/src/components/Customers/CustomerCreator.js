import React, { useState } from 'react'
import Axios from "axios";


import '../Business/businessCreator.sass'


function CustomerCreator() {
    
    // The useState hook that will store the Customer information
    const [customerInformation, setCustomerInformation] = useState([{
        "user_id": 0,
        "custname": ""}]);
        
        let user_id = localStorage.user_id;

    //function that handles Customer creation via axios POST
    // needs to ADD ACTUAL User number once we have it stored from auth0

    // this does seem to be adding a new customer to the backend, but the
    // cust name shows as null, backend function for adding customer info
    // needs to be updated to actually take the info and input it to DB :)
    let postCustomerHandler = () => {
        Axios.post(`${process.env.REACT_APP_BACKEND_URL}users/${user_id}/customers`, customerInformation)
            .then(res => {
                    console.log(res)
                    console.log("HERE customer")
                    localStorage.setItem("customer_id", res.data.id)
                    localStorage.removeItem("business_id")
                }).then(res =>
                    { console.log("I AM HERE customer")
                    window.location.href = '/'
                }
                )
            .catch(error =>{
                console.log('ERROR POST\n',error);
        });
    }

        
    const changeHandler = event => {
        setCustomerInformation({ ...customerInformation, [event.target.name]: event.target.value });
    };
    
    //   submit form function
    const submit = e =>{
        e.preventDefault()
        // 
        if(customerInformation.custname) {
            console.log('do the next thing now ;p', customerInformation.custname)
            postCustomerHandler();
        }
        
        console.log('data to be sent to backend', customerInformation)
        
    }


    //   JSX for customerCreator component
    return (
        <>
    <h3>Create your customer</h3>
    <div className="form"> 
        <form onSubmit={submit}>

            <div className="input-box-type1">
                <label>Display name: <span className="required-span">*</span></label>
                <input
                    type="text"
                    name="custname"
                    value={customerInformation.custname}
                    placeholder="customer name..." 
                    onChange={changeHandler}
                    />
            </div>
            <br/>
            
            <span className="required-span">* required</span>


            <button className="create-business-button"> Create Customer Account </button>

        </form>
    </div>
    </>
    )
}

export default CustomerCreator
