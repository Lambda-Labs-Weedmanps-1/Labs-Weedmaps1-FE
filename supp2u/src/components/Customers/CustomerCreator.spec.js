import React from 'react';
import { mount, configure, shallow, enzyme } from 'enzyme';
import Enzyme from 'enzyme';
import { MemoryRouter } from 'react-router';
import Adapter from 'enzyme-adapter-react-16';

import CustomerCreator from './CustomerCreator'

configure({ adapter: new Adapter() });

describe('AddCustomer Renders', () => {
    // If Map component render is ever non functional, this should fail
    it('should render correctly when component called', () => {
      const Customer = shallow(<CustomerCreator />);

      expect(Customer).toMatchSnapshot();
    });
});

describe('input values for customer creation should be present', () => {
    const container = shallow(<CustomerCreator />)

    it('onchange should work', () => {
        let container = shallow(<CustomerCreator />)
        container.find('input[name="name"]').simulate('change', {
            target: {
                value:'testpassword'
            }
        });
        expect(container.find('input[name="name"]').prop('value')).toBe(
            'testpassword'
        );
    });

    it('should have a input field', () => {
        expect(container.find('input[type="text"]').length).toEqual(1)
    })
});