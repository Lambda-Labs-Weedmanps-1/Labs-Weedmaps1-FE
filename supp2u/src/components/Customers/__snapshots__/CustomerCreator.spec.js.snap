// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCustomer Renders should render correctly when component called 1`] = `
<Fragment>
  <h3>
    Create your customer
  </h3>
  <div
    className="form"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="input-box-type1"
      >
        <label>
          Display name: 
          <span
            className="required-span"
          >
            *
          </span>
        </label>
        <input
          name="custname"
          onChange={[Function]}
          placeholder="customer name..."
          type="text"
        />
      </div>
      <br />
      <span
        className="required-span"
      >
        * required
      </span>
      <button
        className="create-business-button"
      >
         Create Customer Account 
      </button>
    </form>
  </div>
</Fragment>
`;
